### ISSUE TITLE
Mail notification for new share generates invalid email from (Error: `Swift_RfcComplianceException`: Address in mailbox given [...] does not comply with RFC 2822, 3.6.2.)

### ISSUE THREAD

**Issue by 0x47** [nextcloud/server#26683]
The code at: https://github.com/nextcloud/server/blob/d2ea068552eb144d491b76e0a0ef8266144dcf45/lib/private/Share20/Manager.php#L925 creates invalid e-mail from addresses if the server is _not_ configured to use fixed custom `mail_from_address`. Error messages are:
```
Error: Swift_RfcComplianceException: Address in mailbox given [...] does not comply with RFC 2822, 3.6.2.
```
As can be seen in 
https://github.com/nextcloud/server/blob/d2ea068552eb144d491b76e0a0ef8266144dcf45/lib/public/Util.php#L291 the parameter `$user_part` is used if `mail_from_address` has not been set for the instance. The `instanceName` given by the `sendMailNotification` function may contain spaces which is not allowed.
https://github.com/nextcloud/server/blob/d2ea068552eb144d491b76e0a0ef8266144dcf45/lib/private/Share20/Manager.php#L917

Please use a `$user_part` similar to the lost password function: https://github.com/nextcloud/server/blob/9de329a4c2327767d86bd7f594b232eb56af0d01/core/Application.php#L68 Here, the `$user_part` makes sense and is not dependent on arbitrary strings defined in the admin section. Maybe use something like `shares-noreply`.

**Comment by MorrisJobke** [comment]
I noticed this on a setup where no email is configured and it took `root@localhost.localdomain`. Setup check was green. Maybe it should also warn about a missing mail setup in there.


cc @kesselb

**Comment by MorrisJobke** [comment]
The response to the POST request to `/shares` was:

```json
{
    "ocs": {
        "meta": {
            "status": "faulure",
            "statuscode": 403,
            "message": "Address in mailbox given [Powered by Company@localhost.localdomain| does not comply with RFC 2822, 3.6.2."
        }
    }
}
```

That email username is coming most likely from the theming as https://github.com/nextcloud/server/issues/21120#issuecomment-759497084

**Comment by MorrisJobke** [comment]
cc @juliushaertl for the theming part

**Comment by 0x47** [comment]
> Setup check was green. Maybe it should also warn about a missing mail setup in there.

In my opinion the issue is not _a missing mail setup_ but simply a bug in the code. Why would you ever use `$instanceName` as the sender of a mail? It can be an arbitrary string with spaces in it.

Having no sender part in the server mail configuration is a perfectly fine use case. It should (and will in at least two cases) use specific sender parts (`noreply@...` and `lostpassword-noreply@...`) for each purpose.

**Comment by MorrisJobke** [comment]
> > 
> 
> In my opinion the issue is not _a missing mail setup_ but simply a bug in the code. Why would you ever use `$instanceName` as the sender of a mail? It can be an arbitrary string with spaces in it.

True point. 👍 That should be fixed.

**Comment by MorrisJobke** [comment]
Fix is in #27199
